File: Ruby/beta/AddWord2.rb
class Arith
  NUMS = ["zero","one","two","three","four","five","six","seven","eight","nine","ten","eleven","twelve","thirteen","fourteen","fifteen","sixteen","seventeen","eighteen","nineteen","twenty",nil,nil,nil,nil,nil,nil,nil,nil,nil,"thirty",nil,nil,nil,nil,nil,nil,nil,nil,nil,"forty",nil,nil,nil,nil,nil,nil,nil,nil,nil,"fifty",nil,nil,nil,nil,nil,nil,nil,nil,nil,"sixty",nil,nil,nil,nil,nil,nil,nil,nil,nil,"seventy",nil,nil,nil,nil,nil,nil,nil,nil,nil,"eighty",nil,nil,nil,nil,nil,nil,nil,nil,nil,"ninety"]
  
  def initialize(first)
    @first = first
  end
  
  def add(second)
    number_to_word(word_to_number(@first)+word_to_number(second))
  end
  
  private 
  
  def word_to_number(word)
    result = 0
    word = word.split(" ")
    (0...word.length).each do |i|
      if word[i] == "hundred"
        result *= 100
      elsif word[i] != "and"
        result += NUMS.index(word[i])
      end
    end
    result
  end
  
  def number_to_word(num)
    result = []
    if (num / 1000).to_i > 0
      result << NUMS[(num / 1000).to_i]
      result << "thousand"
      num -= (num / 1000).to_i * 1000
    end
  
    if (num / 100).to_i > 0
      result << NUMS[(num / 100).to_i]
      result << "hundred"
      num -= (num / 100).to_i * 100
    end

    if num > 20
      result << "and" if result.length != 0
      if (num / 10).to_i > 0
        result << NUMS[(num / 10).to_i*10]
        num -= (num / 10).to_i * 10
        result << NUMS[(num / 1).to_i] if num != 0
      end
    else
      if num != 0
        result << "and" if result.length != 0
        result << NUMS[num.to_i]
      end
    end
    result.join(" ")
  end
  
end

File: Python/Beta/AddWord2.py
class Arith():
    def __init__(self, first):
        self.NUMS = ["zero","one","two","three","four","five","six","seven","eight","nine","ten","eleven","twelve","thirteen","fourteen","fifteen","sixteen","seventeen","eighteen","nineteen","twenty","","","","","","","","","","thirty","","","","","","","","","","forty","","","","","","","","","","fifty","","","","","","","","","","sixty","","","","","","","","","","seventy","","","","","","","","","","eighty","","","","","","","","","","ninety"]
        self.first = first
        
    def add(self, second):
        return self.number_to_word(self.word_to_number(self.first) + self.word_to_number(second))
    
    def word_to_number(self, word):
        result = 0
        word = word.split(" ")
        for i in range (0,len(word)):
            if word[i] == "hundred":
                result *= 100
            elif word[i] != "and":
                result += self.NUMS.index(word[i])
        return result
        
    def number_to_word(self, num):
        result = []
        if int(num / 1000) > 0:
            result.append(self.NUMS[int(num / 1000)])
            result.append("thousand")
            num -= int(num / 1000) * 1000
  
        if int(num / 100) > 0:
            result.append(self.NUMS[int(num / 100)])
            result.append("hundred")
            num -= int(num / 100) * 100

        if num > 20:
            if len(result) != 0: result.append("and")
            if int(num / 10) > 0:
                result.append(self.NUMS[int(num / 10)*10])
                num -= int(num / 10) * 10
                if num != 0: result.append(self.NUMS[int(num / 1)])
        else:
            if num != 0:
                if len(result) != 0: result.append("and")
                result.append(self.NUMS[int(num)])

        return " ".join(result)